src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift

```diff
@@ -77,11 +79,13 @@ final class DiaryOverviewDayCellModel {
 
 	var exposureHistoryDetail: String? {
 		switch historyExposure {
-		case .encounter:
-			return selectedEntries.isEmpty ?
-				AppStrings.ContactDiary.Overview.riskTextStandardCause :
-				[AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
-
+		case let .encounter(risk):
+			switch risk {
+			case .low:
+				return selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextStandardCause : [AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
+			case .high:
+				return computeExposureHistoryDetail()
```

Comment:
You could move all this logic here. I don't really like the naming `compute...`

```suggestion
			case .high where minimumDistinctEncountersWithHighRisk > 0:
				return diaryDay.selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextStandardCause : [AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
							case .high where minimumDistinctEncountersWithHighRisk >= 0:
				return diaryDay.selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextLowRiskEncountersCause : [AppStrings.ContactDiary.Overview.riskTextLowRiskEncountersCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
```

Reply:
Fixed